src/main/java/network/brightspots/rcv/BaseCvrReader.java

```diff
@@ -32,6 +34,16 @@ abstract class BaseCvrReader {
     this.cvrPath = config.resolveConfigPath(source.getFilePath());
   }
 
+  // Gather candidate names from the CVR that are not in the config.
+  Set<String> gatherUnknownCandidateNames() throws CastVoteRecord.CvrParseException, IOException {
+    try {
+      readCastVoteRecords(new ArrayList<>(), new HashSet<>());
+    } catch (TabulatorSession.UnrecognizedCandidatesException unrecognizedCandidates) {
```

Comment:
Hmm. This is clever! But yeah, I think you're probably right that it's not really a kosher way to use exceptions. Could you just access the list of unrecognized candidates after calling `readCastVoteRecords` without using exception handling? Or maybe just have the parser expose a new method `getUnrecognizedCandidates` or whatever.

Reply:
It may be difficult to handle many types of errors in that case in a way that's resilient to future errors being added?

```
try {
  readCastVoteRecords(...)
} catch(ExceptionType0 e) {
} catch(ExceptionType1 e) {
} catch(ExceptionType2 e) {
}
```

to

```
readCastVoteRecords(...)
getErrorsType0(..)
getErrorsType1(..)
getErrorsType2(..)
```

Or, have all possible errors inherit from some base class, and return a list of that class' type?

A third option that feels the cleanest, but is perhaps the most work, is to create a `CVRParseResults` class:

```
class CVRParseResults() {
  List<CastVoteRecord> CastVoteRecords;
  bool HasErrors;
  Map<String, Integer> UnrecognizedCandidateCounts;
  ...
}
```

And rather than throwing any exceptions, we just check `HasErrors` here.